### PR TITLE
Refactored session handling

### PR DESCRIPTION
**Pull request by timidak**
- Removed session handling from components
- Moved it to `useApollo` hook
- Added Alert in case of invalid token

Fixes
- https://github.com/corona-school/project-user/issues/578
- https://github.com/corona-school/project-user/issues/584

### CODE REVIEW

File: src/pages/Login.tsx [hunk 1]
```diff
@@ -68,6 +60,29 @@ export default function Login() {
 
     useEffect(() => {
         if (sessionState === 'logged-in') navigate(retainPath);
```
Comment (Jonasdoubleyou) [inline src/pages/Login.tsx:62]: When I navigate directly to /login while being logged in, retainPath is undefined and navigate(...) fails:
image

I guess we should assume that retainPath is optional and add a `?? '/start'` here

Comment (Jonasdoubleyou) [reply src/pages/Login.tsx:62]: Looks good apart from this

Comment (Jonasdoubleyou) [reply src/pages/Login.tsx:62]: Uh, we also make the same assumption that the state is set in other places, i.e. https://github.com/corona-school/user-app/blob/8b8bba5f9bc75cd25a935bcd3cb8f8818691c59d/src/pages/Welcome.tsx#L18 and this already affected 20 users:
image

Comment (Jonasdoubleyou) [reply src/pages/Login.tsx:62]: Can you directly fix this in all places tomorrow?

Comment (timidak) [reply src/pages/Login.tsx:62]: Sure, I'll fix them tomorrow in this PR, @Jonasdoubleyou. 🙂

**Review by LomyW — APPROVED** [review]
Looks good 🎉 and works smoothly!

**Comment by Jonasdoubleyou** [comment]
I merged this to get the fix in ASAP ...

**Pull request merged**